### PR TITLE
PE-8289: Add Unit Testing

### PR DESCRIPTION
**Pull request by onwsk8r**

### CODE REVIEW

File: pkg/gmaw/gmaw_test.go [hunk 1]
```diff
@@ -0,0 +1,32 @@
+package gmaw_test
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/juju/errors"
+
+	"github.com/google/go-cmp/cmp"
```
Comment (robloxrob) [inline pkg/gmaw/gmaw_test.go:11]: I like this library ❤️

**Review by robloxrob — APPROVED** [review]
Awesome work. You put the T back in Testing!

**Comment by robloxrob** [comment]
@onwsk8r @mkohn Can we get this pulled in if the work is completed?

**Comment by onwsk8r** [comment]
@robloxrob This is pretty scant coverage - we could finish the `TestMachine_*` functions and use those as a template for other types. Does that make sense (cc @mkohn)?

**Pull request merged**